heron/spi/src/java/com/twitter/heron/spi/statemgr/IStateManager.java

```diff
@@ -191,4 +191,39 @@
    */
   ListenableFuture<PhysicalPlans.PhysicalPlan> getPhysicalPlan(
       WatchCallback watcher, String topologyName);
+
+  /**
+   * Checks whether the tmaster location for the given topology exists
+   *
+   * @return Boolean
+   */
+  ListenableFuture<Boolean> doesTMasterLocationExists(String topologyName);
+
+  /**
+   * Checks whether the scheduler location for the given topology exists
+   *
+   * @return Boolean
+   */
+  ListenableFuture<Boolean> doesSchedulerLocationExists(String topologyName);
```

Comment:
open question, but do we need 'does'? in these methods (e.g., `boolean schedulerLocationExists(..)`).